### ISSUE TITLE
ES6 component fails to compile when object properties shorthand 

### ISSUE THREAD

**Issue by matteofigus** [opentable/oc#358]
server.js is 
```js
'use strict';

module.exports.data = (context, callback) => {
  const username = context.params.username || 'matteo';
  callback(null, { username });
};
```
Related to #346 

This is running the latest version of OC (0.34.0) with node v4.7.2

```
an error happened while packaging /Users/mfigus/Documents/ot/test/es6: ./es6/server.js
Module build failed: SyntaxError: Unexpected token (7:28)
    at Parser.pp.raise (/Users/mfigus/Documents/ot/oc/node_modules/falafel/node_modules/acorn/dist/acorn.js:1745:13)
    at Parser.pp.unexpected (/Users/mfigus/Documents/ot/oc/node_modules/falafel/node_modules/acorn/dist/acorn.js:2264:8)
    at Parser.pp.parsePropertyValue (/Users/mfigus/Documents/ot/oc/node_modules/falafel/node_modules/acorn/dist/acorn.js:1344:15)
    at Parser.pp.parseObj (/Users/mfigus/Documents/ot/oc/node_modules/falafel/node_modules/acorn/dist/acorn.js:1311:10)
    at Parser.pp.parseExprAtom (/Users/mfigus/Documents/ot/oc/node_modules/falafel/node_modules/acorn/dist/acorn.js:1124:19)
    at Parser.pp.parseExprSubscripts (/Users/mfigus/Documents/ot/oc/node_modules/falafel/node_modules/acorn/dist/acorn.js:1023:19)
    at Parser.pp.parseMaybeUnary (/Users/mfigus/Documents/ot/oc/node_modules/falafel/node_modules/acorn/dist/acorn.js:1004:19)
    at Parser.pp.parseExprOps (/Users/mfigus/Documents/ot/oc/node_modules/falafel/node_modules/acorn/dist/acorn.js:948:19)
    at Parser.pp.parseMaybeConditional (/Users/mfigus/Documents/ot/oc/node_modules/falafel/node_modules/acorn/dist/acorn.js:930:19)
    at Parser.pp.parseMaybeAssign (/Users/mfigus/Documents/ot/oc/node_modules/falafel/node_modules/acorn/dist/acorn.js:908:19)
```

**Comment by nickbalestra** [comment]
Thanks @matteofigus .
There are a couple of thing causing this, that I'm currently invesigating:

1. falafel-loader -> acorn 
2. webpack uglify plugin

for `1` the webpack loader I've built should probably fix it
for `2` we have two options as we also discussed: avoid uglifying for the server, or set the babel-preset-env to output to node 0.10 so that everything should be transpiled... I'll personally go with the first of the two options as I'll favor a bit more payload for server.js file, vs babel messing up with a ton of stuff that is not needed (since we dropped 0.10 support). Thoughts?
